src/compute/src/render/top_k.rs

```diff
@@ -197,6 +197,10 @@ where
                     // a single topk stage.
                     let (result, errs) =
                         self.build_topk_stage(thinned, order_key, 1u64, 0, limit, arity, false);
+                    // Consolidate the output of `build_topk_stage` because it's not guaranteed to be.
+                    let result = result.consolidate_named::<KeyBatcher<_, _, _>>(
+                        "Monotonic TopK final consolidate",
+                    );
```

Comment:
The reason I added the defensive consolidate is that I don't know what expectations downstream operators have about the form of the data. Ideally, all operators should function with non-consolidated data, but specifically monotonic implementations do not handle non-consolidated data well. (Should we have a different `Diff` for monotonic dataflows?)

Reply:
Monotonic operators have a `must_consolidate` flag, which informs them whether the input is consolidated.

This is relaxed by `RelaxMustConsolidate`. This does abstract interpretation on the LIR trees, keeping track of whether there was an operation that changed the consolidatedness. You can control its behavior for TopK [here](https://github.com/MaterializeInc/materialize/blob/9cb95d5bad7fd7b5e7213a0ed9674ac775cd951f/src/compute-types/src/plan/interpret/physically_monotonic.rs#L198).

Reply:
Thank you, so the implementation is correct wrt to the physically monotonic interpreter. This is somewhat fragile because reasoning about whether a certain operator is monotonic or not is not simple...

Reply:
Yeah, this interpreter has to be kept up-to-date whenever any operator implementation changes. This might indeed be a little error-prone.